functions/streamtoidv/main.py

```diff
@@ -1,13 +1,58 @@
 import base64
 import json
+import logging
+import utils
 
-print('Loading function')
-
+from cis import validation, encryption
 
 def handle(event, context):
-    print("Received event: " + json.dumps(event, indent=2))
+
+    # Initialize Stream Logger
+    # Log level can be environment driven later in development.
+    log_level = logging.INFO
+    utils.set_stream_logger(level=log_level)
+    logger = logging.getLogger('cis-validator')
+    logger.info("Stream Processor initialized.")
+
     for record in event['Records']:
         # Kinesis data is base64 encoded so decode here
-        payload = base64.b64decode(record['kinesis']['data'])
-        print("Decoded payload: " + payload)
+        logger.info("Record is loaded.")
```

Comment:
if there is no record identifier from kinesis (i dont know the exact format) then that would not be possible. if the record is the actual user profile, we could log record['kinesis']['data']['user_id'] (and yes via format string)